File: solutions/c/sum-of-multiples/1/sum_of_multiples.c
#include "sum_of_multiples.h"

unsigned int sum(const unsigned int *factors, const size_t number_of_factors,
                 const unsigned int limit){
    unsigned int answer=0;
    for (unsigned int i=1;i<limit;i++) {
        for (unsigned int j=0;j<number_of_factors;j++) {
            if (*(factors+j)>0){
                if (i%*(factors+j)==0) {
                    answer=answer+i;
                    j=number_of_factors;
                }
            }
        }
    }
    return answer;
                 }

File: solutions/python/sum-of-multiples/1/sum_of_multiples.py
def sum_of_multiples(limit, factor):
    sum_of_multiples=0
    if len(factor)> 0:
        multiples = []
        for f in factor:
            if f !=0:
                multiples.append([x for x in range(1,limit) if x%f==0])
            else:
                multiples.append([0])
        for f in range(1,len(multiples)):
            multiples[0]=multiples[0]+multiples[f]
        multiples[0]=list(set(multiples[0]))
        sum_of_multiples=sum(multiples[0])
    return sum_of_multiples